src/families/family.h

```diff
@@ -55,6 +55,34 @@ class Family {
   template <typename T>
   mat hessian(const T& x, const mat& y, const mat& lin_pred)
   {
+    if (name() == "multinomial"){
+      mat tmp;
+      tmp = x.t();
+
+      vec lp_max = max(lin_pred, 1);
+      vec lse = trunc_log(exp(-lp_max) 
+                + sum(trunc_exp(lin_pred.each_col() - lp_max), 1)) + lp_max;
+      
+      mat prob_pred = trunc_exp(lin_pred.each_col() - lse);
+
+      uword p = x.n_cols;
+      uword m = y.n_cols;
+      uword n = y.n_rows;
+
+      mat H(p*m, p*m, fill::zeros);
+
+      for (uword i = 0; i < m; i++) {
+        for (uword j = 0; j < m; j++) {
+          for (uword k = 0; k < n; k++) {
+            vec v = tmp.col(k);
+            H(span(i*p, (i+1)*p -1), span(j*p, (j+1)*p -1)) +=
+              prob_pred(k, i)*((i == j) - prob_pred(k, j))*(v*v.t());
+          }
+        }
+      }
+      
+      return H;
+    }
```

Comment:
Why don't you code this up as a method for the multinomial class?

Reply:
Could you suggest on a design for this? Simple overriding "hessian()" won't work because bfgs() function is a member of base class 'family' and the 'hessian()' call inside it calls 'Family::hessian()' instead of the overriding function "Multinomial::hessian()". 
The only solution that came in my mind was writing getting rid of pseudoHessian and expliticitly writing hessian for each derived class.

Reply:
Another way would be to imitate how ADMM is implemented in SLOPE's code. Declaring separate functions to get rid of template argument and then declaring both of them as virtual functions.

Reply:
You're right, I see the problem and have run into it several times before. Let's just keep it as it is for now.

Sometimes I wonder if it would just be simpler to use branching instead of classes for these functions.